### PR TITLE
fix: safer fetch plus error reporting

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [PHIRE-325] 

### Description

We want to have some more visibility on what might go wrong on the places were we use fetch thus we added some `try/catch` blocks around the places where we use fetch.

Started with implementing a wrapper called safeFetch around fetch but didn't work in all of our cases so I removed it in the end.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- safer fetch plus error reporting - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-325]: https://artsyproduct.atlassian.net/browse/PHIRE-325?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 24e55fe696aa3a1c197c4a8c04a3127490e26db0
</p>

File: src/app/Components/ShareSheet/helpers.ts [hunk 1]
```diff
@@ -64,3 +66,23 @@ export const getShareImages = (shareSheetItem: ShareSheetItem) => {
     currentImageUrl,
   }
 }
+
+export async function getImageBase64(url: string): Promise<string> {
```
Comment (araujobarret) [inline src/app/Components/ShareSheet/helpers.ts:70]: 💟

**Review by araujobarret — APPROVED** [review]
💪

**Review by brainbicycle — COMMENTED** [review]
awesome!

**Pull request merged**